### ISSUE TITLE
Funkstagram issues

### ISSUE THREAD

**Issue by boxercox** [jrobson153/funkstagram#4]
The plugin is being used on my site to allow my client to take images and upload them to their media library easily so from there they can easily add them to products in their woo commerce store.

The plugin doesn't seem to be automatically adding them in and only adds in some images when I click the import now button.

Initially it was working fine but now its very intermittant.
I have a very tech unsavvy client so I set this up to make their product set ups very easy.

We also have a pro instagram feed on the site and a instagram autoposter to post final products in the store to their live instagram account.

Hoping you can help to trouble shoot why its not loading these images into the library.

Site is using latest Wordpress and current version of the Salient theme with rolled back woo to accomodate Salient

**Comment by johndigital** [comment]
Hey @boxercox sorry about the issues. This one might be tough to troubleshoot, here are a few questions I have off the bat:

- Do you have the latest version installed from this repo?
- Are you using hashtags or only importing by username?
- Are you trying to import images from multiple users or only one?
- Just to re-iterate, the images do get imported correctly and end up in your media library when you use the "import now" button?
- Does the plugin report any errors when using the "import now" button?
- Is the account you've authenticated the same as the one you used to set up the app in the IG developer dashboard?

Thanks!

**Comment by boxercox** [comment]
> On 13 Feb 2018, at 1:32 am, John Robson <notifications@github.com> wrote:
> 
> Hey @boxercox <https://github.com/boxercox> sorry about the issues. This one might be tough to troubleshoot, here are a few questions I have off the bat:
> 
> Do you have the latest version installed from this repo?

Yes I think so. Can you point me to where to download it again in case?
> Are you using hashtags or only importing by username?
Just one username
> Are you trying to import images from multiple users or only one?
only one
> Just to re-iterate, the images do get imported correctly and end up in your media library when you use the "import now" button?
Intermittantly … a whole chunk of them didn’t auto import and it was only brought to my attention 2 days later so I then tried to import via the button but they didn’t come through. The next couple of images I took I then hit the button and they imported.. I have been checking in and I can see new images in there but I believe my client is also using the button to “make sure"
> Does the plugin report any errors when using the "import now" button?

No none

> Is the account you've authenticated the same as the one you used to set up the app in the IG developer dashboard?

?
Yes I used the button to set up a key for this specific account

**Comment by boxercox** [comment]
Hi I eventually got this working and then it completely stopped again. Now I get an error that says:


3 errors returned:
0 new images found and uploaded

1.    User vae807 does not exist
2.    User Vart does not exist
3.    Must set at least 1 user or tag

Whilst that user most definitley DOES EXIST  ie

https://www.instagram.com/vae807/

What can I do...

**Comment by boxercox** [comment]
John can you help?

**Comment by drewbaker** [comment]
@boxercox I would expect this to not work, after all the changes to the Instagram API from a few weeks ago. Instagram/Facebook has basically disabled all the APIs. There really isn't a good way to access content that isn't "yours" via official APIs anymore.

**Comment by boxercox** [comment]
The content IS ours. We Use it to simply create our images for our shop and have them upload directly into our media library. Its the best way for the shop owner to create content ready imagery as she isn'r very web savvy. She can't even work out how to manually upload images let alone how to down load images from her own instagram.

Can you give me any more suggestions?  PS finding anyone to ask about this plugin is really hard as the back end of the plugin just points to your commercial site

**Comment by boxercox** [comment]
Also I AM affectively using instagram feeds in lots of other places on my site. MY IMAGES are being fed in using Instagram feed pro and are working fine. And also I am using another plugin calld Instagra Autoposter and this too is working well.

I now just need one to work back the other way.. where I can post to my other instagram account and have the images load to my media library so i can sue them in products listings and blog posts.

**Comment by drewbaker** [comment]
We built this plugin for our clients sites and open sourced it for developers, it's certainly not something that you should expect to have any sort of commercial grade support around. 

When I say the content "isn't yours" I mean, the Instagram API key needs to belong to the user who's imagery you are trying to access. 

This is some more background about what we are up against now: https://techcrunch.com/2018/04/02/instagram-api-limit/

**Comment by johndigital** [comment]
Hey @boxercox,

Yea, as @drewbaker mentioned if this was working for you and is not any longer it is likely due to the [massive API changes](https://techcrunch.com/2018/04/04/facebook-instagram-api-shut-down/) that Instagram dropped on everyone a few weeks ago. We've been dealing with the consequences of those changes in all sorts of ways, but supporting this plugin is not something we're planning to pursue anymore.

I know it will probably take some re-working for you, but our [social-pull plugin](https://github.com/funkhaus/social-pull) can probably solve the problem you're having. It's not very well documented at the moment, but it is working currently and since it relies on [Zapier](https://zapier.com/) to do the actual polling there is much less functionality that we have to support moving forward.

To get it working you would just install it on your site, follow the readme to set up a token in the settings page, and generate the webhook URL. Once you have that you just head over to [Zapier](https://zapier.com/) and set up a new zap to import instagram posts. You set the "action" end of the zap to be their webhook service, and point it to the URL that you generated in the social-pull backend sending all the data as JSON.

From there, all instagram posts will end up in your media library and will additionally be represented as posts in the "sp-social" custom post type. 

Hope that helps you in some way, but again as @drewbaker mentioned these plugins are internal tools for our agency and though they are open source we do not often have the bandwidth to support all of them for everyone.

**Comment by boxercox** [comment]
OK I will take a look. I haven't used zapier before ( I tend to use IFTTT) .

FYI the instagram account we are using IS ours and we set up the api key so shouldn't be any issues at all there.

Thanks for your responses. I will give this a whirl.

**Comment by boxercox** [comment]
One last thing.

Can I upload the plugin via the plugin installer on wp ( will it set up the appropriate folder in wp-content/plugins I don't currently have access to the cp

AND... in terms of a Random ASCI do I use this to generate one? and how many characters will it need?
https://www.random.org/strings/

**Comment by boxercox** [comment]
Eek  this happened...


Fatal error: Cannot redeclare funkstagram_admin_style() (previously declared in /home/vintageart/public_html/wp-content/plugins/funkstagram/funkstagram-settings.php:7) in /home/vintageart/public_html/wp-content/plugins/social-pull-master/settings.php on line 13

**Comment by johndigital** [comment]
Sounds good, Zapier is very similar to IFTTT

Yes, the plugin installer should work fine. To get rid of that error just disable and/or remove funkstagram. 

The ASCI string can be anything at all (as long as its URL safe characters), that generator should work fine. It's just a security precaution to prevent any 3rd party from being able to use the webhook endpoint to inject content to your site.

**Comment by boxercox** [comment]
I think I broke my site https://www.vintageartemporium.com/wp-admin/plugins.php

**Comment by boxercox** [comment]
Got it all installed and set up Zapier  but looks like social pull isn't sure what to do.

I need my images from instagram to appear in my medialibrary. I have the webhook working in Zapier.

but I am trying to get an image posted on Instagram to be added to the WordPress media library using Zapier.

I have configured Social Pull in Wordpress, and set up a zap in Zapier.

The tests in zapier show as being successful, but the latest Instagram image is not added to WordPress.

Can you please let me know how I need to configure the Zapier Webhook template so that the Instagram image is sent into the Wordpress media library.
